python/dgl/utils.py

```diff
@@ -155,18 +187,23 @@ def get_items(self, index):
             # the provided index is not a slice
             tensor = self.tousertensor()
             index = index.tousertensor()
-            return Index(F.gather_row(tensor, index))
+            # TODO(Allen): Change F.gather_row to dgl operation
+            return Index(F.astype(F.gather_row(tensor, index),
+                                  F.data_type_dict[self.dtype]), self.dtype)
```

Comment:
Is `F.astype` needed? The result should have the same dtype as `tensor` right?

Reply:
Removed